### PR TITLE
Inline mdi icons with babel plugin + bump build deps

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Bump webpack, babel and corejs

Add new babel plugin to replace imports with the value that would be imported

Use that plugin for mdi icons

Optimize build by caching babel output in dev mode

Fix stats build chunk names, change production chunk names

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: build-scripts/bundle.js [hunk 1]
```diff
@@ -56,12 +56,23 @@ module.exports.babelOptions = ({ latestBuild }) => ({
...
   ].filter(Boolean),
   plugins: [
+    [
+      path.resolve(
+        paths.polymer_dir,
+        "build-scripts/babel-plugins/inline-constants-plugin.js"
+      ),
+      {
+        modules: ["@mdi/js"],
+        ignoreModuleNotFound: true,
```
Comment (balloob) [inline build-scripts/bundle.js:73]: Why should we not raise here?

File: build-scripts/bundle.js [hunk 2]
```diff
@@ -74,8 +85,14 @@ module.exports.babelOptions = ({ latestBuild }) => ({
     "@babel/plugin-proposal-nullish-coalescing-operator",
     ["@babel/plugin-proposal-decorators", { decoratorsBeforeExport: true }],
     ["@babel/plugin-proposal-private-methods", { loose: true }],
+    ["@babel/plugin-proposal-private-property-in-object", { loose: true }],
     ["@babel/plugin-proposal-class-properties", { loose: true }],
   ].filter(Boolean),
+  exclude: [
+    // \\ for Windows, \/ for Mac OS and Linux
```
Comment (balloob) [inline build-scripts/bundle.js:92]: Comment doesn't match regex

**Review by balloob — DISMISSED** [review]
Nice!

File: build-scripts/bundle.js (hunk 1 above)
Comment (bramkragten) [reply build-scripts/bundle.js:73]: Because babel can't handle imports without extension, and most of those are not important (at least for now)

**Review by balloob — APPROVED** [review]

**Comment by steverep** [comment]
@bramkragten what was the goal of replacing the icon imports with constants?  Were the unused ones not being tree shaken out?

**Pull request merged**